### PR TITLE
[MapView][Part 2] Make LocationManager hold LocationOptions directly

### PR DESCRIPTION
**Pull request by nishant-karajgikar**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [x] Document any changes to public APIs.
 - [ ] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-ios` changelog: `<changelog></changelog>`.
 - [ ] Update the migration guide, API Docs, Markdown files - Readme, Developing, etc

### Summary of changes

LocationManager now holds the `LocationOptions` directly similar to gestures.

The call site is now:

```swift
mapView.location.options.puckType = .puck2D()
```

### CODE REVIEW

**Review by jmkiley — APPROVED** [review]

File: Sources/MapboxMaps/Location/LocationManager.swift [hunk 1]
```diff
@@ -39,12 +39,25 @@ public class LocationManager: NSObject {
     /// Only created if `showsUserLocation` is `true`
     internal var locationPuckManager: LocationPuckManager?
 
-    public private(set) var locationOptions: LocationOptions
+    public var options = LocationOptions() {
+        didSet {
+
```
Comment (neelmistry94) [inline Sources/MapboxMaps/Location/LocationManager.swift:44]: nit: remove new line

File: Sources/MapboxMaps/Location/LocationManager.swift [hunk 2]
```diff
@@ -39,12 +39,25 @@ public class LocationManager: NSObject {
...
+            locationProvider.locationProviderOptions = options
+
+            if options.puckType != oldValue.puckType {
+                syncUserLocationUpdating()
+            }
+
+            if let puckType = options.puckType, puckType != oldValue.puckType {
+                locationPuckManager?.changePuckType(to: puckType)
+            }
+
```
Comment (neelmistry94) [inline Sources/MapboxMaps/Location/LocationManager.swift:56]: nit: remove new line

**Review by neelmistry94 — APPROVED** [review]

File: Sources/MapboxMaps/Location/LocationManager.swift (hunk 1 above)
Comment (nishant-karajgikar) [reply Sources/MapboxMaps/Location/LocationManager.swift]: done.

File: Sources/MapboxMaps/Location/LocationManager.swift (hunk 2 above)
Comment (nishant-karajgikar) [reply Sources/MapboxMaps/Location/LocationManager.swift]: done

**Pull request merged**